### PR TITLE
Inventory slots will dim when unusable due to equipment or bodyparts, rather than just for species hardcoded slots

### LINKED ISSUE #54039: Nightmares have blocked jumpsuit slot and no integrated jumpsuit a-la golems, but not blocked jumpsuit-requiring slots
Means they have visibly available pockets, ID, and belt but can't put anything there.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Closes #54039

Rather than only dimming inventory slots your species are unable to use, unusable inventory slots in general are dimmed.
For example, not wearing a jumpsuit shows this:

image

## Why It's Good For The Game

Makes it a bit more transparent what you can and can't wear given your current equipment or status. 

## Changelog

:cl: Melbert
qol: Unusable inventory slots (as according to your bodyparts or equipment) will now be dimmed (similarly to how unusable inventory slots are dimmed for certain species like Golems).
/:cl:

### CODE REVIEW

**Comment by Sealed101** [comment]
hello
https://github.com/tgstation/tgstation/issues/54039

**Comment by MrMelbert** [comment]
zoop 
image

**Comment by ZephyrTFA** [comment]
Can you make it have an overlay with a symbol of some form also. 
Maybe a padlock or large X

**Comment by MrMelbert** [comment]
I'm not gonna touch it further I'll leave it up to the spriters

File: code/_onclick/hud/human.dm [hunk 1]
```diff
@@ -306,16 +306,40 @@
...
+				blocked_slots |= ITEM_SLOT_ID|ITEM_SLOT_BELT
+			var/obj/item/bodypart/left_leg = human_mob.get_bodypart(BODY_ZONE_L_LEG)
+			if(isnull(left_leg) || IS_ORGANIC_LIMB(left_leg))
+				blocked_slots |= ITEM_SLOT_LPOCKET
+			var/obj/item/bodypart/right_leg = human_mob.get_bodypart(BODY_ZONE_R_LEG)
+			if(isnull(right_leg) || IS_ORGANIC_LIMB(right_leg))
+				blocked_slots |= ITEM_SLOT_RPOCKET
+		if(isnull(human_mob.wear_suit))
+			blocked_slots |= ITEM_SLOT_SUITSTORE
+		if(human_mob.num_hands < 2) // update this when you can wear gloves on one hand
```
Comment (mc-oofert) [inline code/_onclick/hud/human.dm:326]: its a thing now

**Comment by LemonInTheDark** [comment]
fucking artisan

**Pull request merged**